### PR TITLE
Issue #SB-5038 feat: form service db migration script

### PR DESCRIPTION
**Pull request by anoophm**

### CODE REVIEW

File: cassandra-migration-etl/form-service-migration/migrate.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+const cassandra = require('cassandra-driver');
+let cassandraClientOptions = { contactPoints: [process.argv[2]] };
+if(process.argv[3] && process.argv[4]){
+  cassandraClientOptions.authProvider = new cassandra.auth.PlainTextAuthProvider(process.argv[3], process.argv[4]);
+}
+console.log('connecting to DB with', process.argv[2], process.argv[3], process.argv[4]);
+const client = new cassandra.Client(cassandraClientOptions);
+
+let transformed_data = [];
```
Comment (bvinayakumar) [inline cassandra-migration-etl/form-service-migration/migrate.js]: why variables are in different conventions?

cassandraClientOptions
transformed_data

File: cassandra-migration-etl/form-service-migration/migrate.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  framework: undefined,
+  type: undefined,
+  subtype: undefined,
+  component: undefined,
+  action: undefined,
+  created: new Date(),
+  last_modified: undefined,
+  data: undefined
+}
+//create keyspace
```
Comment (bvinayakumar) [inline cassandra-migration-etl/form-service-migration/migrate.js]: create key space? remove the comment as it seems irrelevant.

File: cassandra-migration-etl/form-service-migration/migrate.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+let cassandraClientOptions = { contactPoints: [process.argv[2]] };
+if(process.argv[3] && process.argv[4]){
+  cassandraClientOptions.authProvider = new cassandra.auth.PlainTextAuthProvider(process.argv[3], process.argv[4]);
+}
+console.log('connecting to DB with', process.argv[2], process.argv[3], process.argv[4]);
+const client = new cassandra.Client(cassandraClientOptions);
+
+let transformed_data = [];
+let dest_obj = {
+  root_org: undefined,
```
Comment (bvinayakumar) [inline cassandra-migration-etl/form-service-migration/migrate.js]: Use camel case for column names. e.g. rootOrgId

Have following columns as well.

createdOn timestamp, 
createdBy text, 
lastUpdatedOn timestamp, 
lastUpdatedBy text

File: cassandra-migration-etl/form-service-migration/migrate.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+          temp.data = row.data;
+          temp.framework = "*";
+          transformed_data.push(temp);
+        }
+      }
+    })
+  })
+  .then(async _ => {
+    // create keyspace if not exist
+    await client.execute('CREATE KEYSPACE IF NOT EXISTS qmzbm_form_service WITH REPLICATION = { \'class\' : \'SimpleStrategy\', \'replication_factor\' : 1 }')
```
Comment (bvinayakumar) [inline cassandra-migration-etl/form-service-migration/migrate.js]: Replication class and factor shouldn't be hardcoded. In multimode setup, the replication could be as follows.

REPLICATION = { 'class' : 'NetworkTopologyStrategy', 'datacenter1' : 1, 'datacenter2': 1 };

File: cassandra-migration-etl/form-service-migration/migrate.js (hunk 3 above)
Comment (sunilpes) [reply cassandra-migration-etl/form-service-migration/migrate.js]: [Cassandra will by default lowercase all identifier names unless quoted](https://docs.datastax.com/en/cql/3.1/cql/cql_reference/ucase-lcase_r.html). So for the column names, i'm keeping it in snake case.

File: cassandra-migration-etl/form-service-migration/migrate.js (hunk 1 above)
Comment (sunilpes) [reply cassandra-migration-etl/form-service-migration/migrate.js]: Will fix this

File: cassandra-migration-etl/form-service-migration/migrate.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
+// Form service migration script
```
[the remark this answers is not in the document]
Comment (sunilpes) [reply cassandra-migration-etl/form-service-migration/migrate.js]: ok

File: cassandra-migration-etl/form-service-migration/migrate.js (hunk 2 above)
Comment (sunilpes) [reply cassandra-migration-etl/form-service-migration/migrate.js]: ok. but no harm :)

File: cassandra-migration-etl/form-service-migration/migrate.js (hunk 4 above)
Comment (sunilpes) [reply cassandra-migration-etl/form-service-migration/migrate.js]: Ok but how do we know the setup is different? I need more info on this. I'm kept the defaults

**Comment by bvinayakumar** [comment]
Closing this pull request as it is not going in release 1.10.
